ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -1421,6 +1438,8 @@ void handleResponse(ResponseInfo responseInfo, PutResponse putResponse) {
                   chunkPutRequestInfo.replicaId, putResponse.getError(), putResponse.getCorrelationId(), blobId);
               processServerError(putResponse.getError());
```

Comment:
yeah, we could but I feel we may not get any benefit by doing that. Distinguishing `OperationDisabled` from `AmbryUnavailable` is to help operation tracker to make decisions only. For external components (netty layer), OperationDisabled is also a kind of AmbryUnavailable. Hence, I probably would not touch processServerError to avoid any logic depending on AmrbyUnavailable router error code.